rmk/src/keymap.rs

```diff
@@ -185,41 +180,17 @@ impl<const ROW: usize, const COL: usize, const NUM_LAYER: usize> KeyMap<ROW, COL
         self.layers[layer_num][row][col]
     }
 
-    /// Fetch the action in keymap, with layer cache
-    pub(crate) fn get_action_with_layer_cache(
-        &mut self,
-        row: usize,
-        col: usize,
-        key_state: KeyState,
-    ) -> KeyAction {
-        if key_state.is_releasing() {
-            // Releasing a pressed key, use cached layer and restore the cache
```

Comment:
Why removing layer cache? Removing it will lead the key is not releasing when switching layers

Reply:
That info is stored in new struct HoldingKey. Just more convenient to have related info stored together, including hold_start. 
Also use less memory. Instead of store it for every location in matrix, it only need space for 16 holding keys.

Reply:
What will happen if user holds more than 16 keys?

Reply:
new press are ignored. Until some keys are released.
How many simultaneously pressed keys do you want?

Reply:
I want to record all pressed keys on the keyboard, since the number of keys varies, a fixed number of pressed key is not good imo.